### PR TITLE
update to jakarta-based dependencies

### PR DESCRIPTION
**Pull request by neowcng**
- migrate to jakarta-based dependencies (e.g. javax.servlet -> jakarta.servlet)
- updated runtime requirement: JDK 17 or above
- removed ehcache v1 support
- add jsr-107 JCache support
- use log4j2-api instead of log4j v1 and commons-logging

### CODE REVIEW

**Comment by afattahi54** [comment]
Greate, will this be merged? When do you think we can use it

File: README.md [hunk 1]
```diff
@@ -45,7 +45,7 @@ If you want to deepen your knowledge and learn about advanced techniques, I sugg
         <dependency>
             <groupId>de.galan.packtag</groupId>
             <artifactId>packtag-core</artifactId>
-            <version>3.13.2</version>
+            <version>4.0.0=SNAPSHOT</version>
```
Comment (galan) [inline README.md:48]:
```suggestion
            <version>4.0.0-SNAPSHOT</version>
```

**Comment by galan** [comment]
Oh wow, so many love for this old library. Thank you :heart: So it is still used on some website? Only found a small typo, would merge and release this soon. (I'm busy with my life, so forgive my delayed responses)

**Comment by neowcng** [comment]
@galan hi can you prepare the release :)

**Comment by afattahi54** [comment]
Will it available on maven repo too ?

**Comment by galan** [comment]
I had some issues due to changed central authorization, but it is [released](https://central.sonatype.com/namespace/de.galan.packtag) now. :tada:

**Pull request merged**